doc/src/manual/types.md

```diff
@@ -1590,15 +1590,64 @@ to adjust printing.
+since the other functions mentioned here (except `write`) call `show`
+in the absence of a more specific method.
+
+* [`show(io, x)`](@ref), with two arguments,
+  is the default simple text representation of `x`.
+  It is typically the format you would employ to input `x` into Julia.
+
+* [`show(io, mime, x)`](@ref), with three arguments,
+  performs verbose pretty-printing of `x`.
+  Multiple 3-argument `show` methods can be defined for various MIME types to enable
+  richer display of `x` in some interactive environments as discussed above.
+  By default (if no 3-argument method is defined for `typeof(x)`),
+  it calls the 2-argument `show(io, x)`.
+
+* [`print(io, x)`](@ref) by default calls `show(io, x)`,
+  but a few types have a distinct `print` format —
+  most notably, when `x` is a string, `print` outputs the raw text
+  whereas `show` outputs an escaped string enclosed in quotation marks.
+  ([`println`](@ref) and [`printstyled`](@ref) both call [`print`](@ref),
+  but add a newline or styling, respectively.)
+
+* [`display(x)`](@ref) tells the current environment to display `x`
+  in whatever way it thinks best.
+  This is the function used by the REPL to output the result of an evaluated expression.
+  In the REPL, `display` calls `show(io, MIME"text/plain", x)`.
+  In a notebook, like Jupyter or Pluto, `display` calls `show(io, MIME"text/html", x)`
+  (or sometimes `show(io, MIME"image/<format>", x)` if `x` is representable as an image).
+
+* [`write(io, x)`](@ref), if it is defined
+  (it generally has *no* default definition for new types),
+  writes a "raw" binary representation of `x` to `io`,
+  e.g. an `x::Int32` will be written as 4 bytes.
+
+The first argument to these functions, `io`, defines where the output should be written.
+In the REPL, `io` is an [`IOContext`](@ref) wrapper around [`stdout`](@ref).
+[`stdout`](@ref) directs output to the default display, typically the terminal window.
+The `io` argument for all the above functions defaults to `stdout` if it is omitted.
+It is also helpful to be familiar with the metadata that can be attached to an `io` stream
+by an [`IOContext`](@ref) wrapper.
+For example,
+the REPL sets the `:limit => true` flag from `display` for an evaluated expression,
+in order to limit the output to fit in the terminal;
+you can query this flag with `get(io, :limit, false)`.
+And when displaying an object contained within, for example, a multi-column matrix,
+the `:compact => true` flag could be set,
+which you can query with `get(io, :compact, false)`.
+
+The functions `string`, `annotatedstring`, and `repr`
+may be used to save values to a string (rather than output to `io`).
+These functions call either `show` or `print` as indicated below,
+so it is usually not necessary to define new methods for them separately.
+Note that [`convert(String, x)`](@ref) is purposely not defined
+so the user can be explicit about the string representation they desire.
```

Comment:
I think that the real reason is that `convert` in Julia is used for *implicit* conversions, e.g. when assigning an expression to an container of a different type.  So it is *never* defined for converting between conceptually different types of objects, e.g. between numbers and strings.